### PR TITLE
feat(query): moved 'Path Parameter Not Required' to general

### PR DESCRIPTION
**Pull request by rafaela-soares**
Signed-off-by: Rafaela Soares <rafaela.soares@checkmarx.com>

**Proposed Changes**
- Moved "Path Parameter Not Required" query to general
- Implemented `walk` function

I submit this contribution under the Apache-2.0 license.

### CODE REVIEW

File: assets/queries/openAPI/general/path_parameter_not_required/query.rego [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+	[path, value] := walk(doc)
+	param := value.parameters[n]
+	param.in == "path"
+	not_required(param)
+
+	result := {
+		"documentId": doc.id,
+		"searchKey": sprintf("%s.parameters.name={{%s}}", [openapi_lib.concat_path(path), param.name]),
+		"issueType": "MissingAttribute",
```
Comment (felipe-avelar) [inline assets/queries/openAPI/general/path_parameter_not_required/query.rego:18]: I think, if `param.required` is undefined, makes sense that attribute is missing, but when `param.required == false` should be categorized as `IncorrectValue`, no?

**Review by felipe-avelar — CHANGES_REQUESTED** [review]
Please, check comments

File: assets/queries/openAPI/general/path_parameter_not_required/query.rego (hunk 1 above)
Comment (rafaela-soares) [reply assets/queries/openAPI/general/path_parameter_not_required/query.rego]: Yes, it makes perfect sense 🙂

**Review by felipe-avelar — APPROVED** [review]
LGTM

**Pull request merged**